setup.py

```diff
@@ -35,6 +38,44 @@ def run(self):
         os.system('rm -vrf ./build ./dist ./*.pyc ./*.tgz ./*.egg-info')
 
 
+def git_version(version):
+    try:
+        import git
+    except ImportError:
+        logger.warn('gitpython not found: Cannot compute the git version.')
+        return ''
+    try:
+        repo = git.Repo('.git')
+    except ImportError:
+        logger.warn('Git repo not found: Cannot compute the git version.')
+        return ''
+    sha = repo.head.commit.hexsha
```

Comment:
Doesn't else make sense if there is a reasonable default as mentioned on http://stackoverflow.com/a/14590478/1110993